internal/pkg/manifest/svc.go

```diff
@@ -62,11 +64,33 @@ type Sidecar struct {
 	Sidecars map[string]*SidecarConfig `yaml:"sidecars"`
 }
 
+// SidecarsOpts converts the service's sidecar configuration into a format parsable by the templates pkg.
+func (s *Sidecar) SidecarsOpts() ([]*template.SidecarOpts, error) {
+	if s.Sidecars == nil {
+		return nil, nil
+	}
+	var sidecars []*template.SidecarOpts
+	for name, config := range s.Sidecars {
+		port, protocol, err := parsePortMapping(config.Port)
+		if err != nil {
+			return nil, fmt.Errorf("parse port mapping %s: %w", *config.Port, err)
```

Comment:
is it possible that `config.Port` is nil and we blow up? also I don't think we need to wrap this error since `parsePortMapping` is private and we return a nice error there

Reply:
> is it possible that `config.Port` is nil and we blow up? 

nope and it'll be defaulted to be 80.

Reply:
done !